src/Operations.jl

```diff
@@ -364,13 +411,41 @@ function update_manifest(env::EnvCache, uuid::UUID, name::String, hash::SHA1, ve
         push!(infos, info)
     end
     info["version"] = string(version)
-    info["hash-sha1"] = string(hash)
+    if isempty(path)
+        @assert hash != nothing
+        haskey(info, "path") && delete!(info, "path")
+        info["hash-sha1"] = string(hash)
+    else
+        haskey(info, "hash-sha1") && delete!(info, "hash-sha1")
+        info["path"] = path
+    end
     delete!(info, "deps")
-    for path in registered_paths(env, uuid)
-        data = load_package_data(UUID, joinpath(path, "dependencies.toml"), version)
-        data == nothing && continue
-        info["deps"] = convert(Dict{String,String}, data)
-        break
+    if !isempty(path)
+        reqfile = joinpath(path, "REQUIRE")
+
+        !isfile(reqfile) && return
+        deps = Dict{String,String}()
+        pkgs = PackageSpec[]
+        for r in filter!(r->r isa Pkg2.Reqs.Requirement, Pkg2.Reqs.read(reqfile))
+            push!(pkgs, PackageSpec(r.package))
+        end
+        # TODO: Get rid of this one?
+        registry_resolve!(env, pkgs)
```

Comment:
How do we determine if this should or shouldn't be gotten rid if?

Reply:
The problem is using REQUIRE files that do not have UUIDs so the only way to get the UUIDS is to go to the registry. When we are not using REQUIRE files the UUIDs can just be read from the Manifest file.

Reply:
This should be harmless – if all the packages are already resolved, it won't hit the registry at all.